### PR TITLE
Use smaller archives for external dependencies

### PR DESCRIPTION
**Pull request by Luthaf**
This reduces the zlib and tng archives, bringing the size of chemfiles archive from 16MiB down to 6MiB. This was prompted by crates.io refusing to upload the crate for chemfiles-sys.

I have also added a test to catch any file size regression.

### CODE REVIEW

**Comment by Luthaf** [comment]
Related PR: https://github.com/chemfiles/tng/pull/2 and https://github.com/chemfiles/zlib/pull/2

File: tests/lints/check-external-size.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    "mmtf-cpp.tar.gz": 439,
+    "tng.tar.gz": 317,
+    "zlib.tar.gz": 370,
+    "molfiles.tar.gz": 477,
+    "netcdf.tar.gz": 494,
+    "xdrfile.tar.gz": 41,
+    "bzip2.tar.gz": 344,
+    "pugixml.tar.gz": 549,
+    "gemmi.tar.gz": 476,
+    "fmt.tar.gz": 745,
```
Comment (ezavod) [inline tests/lints/check-external-size.py:28]: Nit: wouldn't it be nice to have things in alphabetical order?
```suggestion
    "bzip2.tar.gz": 344,
    "fmt.tar.gz": 745,
    "gemmi.tar.gz": 476,
    "lzma.tar.gz": 256,
    "mmtf-cpp.tar.gz": 439,
    "molfiles.tar.gz": 477,
    "netcdf.tar.gz": 494,
    "pugixml.tar.gz": 549,
    "tng.tar.gz": 317,
    "xdrfile.tar.gz": 41,
    "zlib.tar.gz": 370,
```

**Review by ezavod — APPROVED** [review]
Awesome, looking forward to have hopefully twice as fast crate downloading in the future.

**Comment by Luthaf** [comment]
> looking forward to have hopefully twice as fast crate downloading in the future.

Unfortunately no, the previous release was already around 5 MB, the jump to 11MB was a regression. However, with https://github.com/chemfiles/chemfiles.rs/commit/dd835d479329d57d8f78af7c41521a224c8ecb14 you should see much faster build times.

**Pull request merged**